mail/src/main/java/com/sun/mail/pop3/Protocol.java

```diff
@@ -461,11 +475,58 @@ boolean authenticate(String host, String authzid,
 		    logger.fine("AUTH " + mech + " command trace suppressed");
 		    suspendTracing();
 		}
-		if (ir != null)
-		    resp = simpleCommand("AUTH " + mech + " " +
-					    (ir.length() == 0 ? "=" : ir));
-		else
-		    resp = simpleCommand("AUTH " + mech);
+		if (ir != null) {
+			resp = simpleCommand("AUTH " + mech + " " +
+					(ir.length() == 0 ? "=" : ir));
+
+			/*
+			 * Fallback for two lines authentication format (e.g Microsoft 365 and Azure)
+			 * in order to establish successful connection. This is something that indeed might be patched on the Microsoft side one day.
+			 * though it is backward compatible and doesn't just come down to Microsoft
+			 * Details:
+			 * 	https://github.com/eclipse-ee4j/mail/issues/461
+			 * 	https://techcommunity.microsoft.com/t5/exchange-team-blog/announcing-oauth-support-for-pop-in-exchange-online/ba-p/1406600
+			 */
+			if (!resp.ok) {
```

Comment:
Could discover this two lines format from properties (speciffic provider) and have dedicated authenticator for it ?

Reply:
I agree it would be worth allowing the client to provide a property which could be used jump straight to the two line auth when provided to improve the performance and avoid a connection failure on each attempt. 

I wouldn't link it to provider here though as in the future Microsoft might support one line auth like Gmail.

Reply:
The property would have to be tied to a specific connection and I find it hard to implement this way. It cannot be a system property as the consumer might have multiple connections which use a one as well as two line authentication format. 

Ideally, it would have to be configured and passed from the consumer while performing queries on behalf of a specific connection. 

 or am I missing something?

Reply:
I believe the properties here are the ones which come from the creation of the Session which the client would create. You create one of these for each mailbox that you connect to and I do not think are tied to system properties by default. Therefore, I think it should be possible to set a mail.pop3s.auth.single.command or similar where the default is true?

Reply:
That makes more sense, I'll give it a try.